CRM/Core/Smarty.php

```diff
@@ -98,16 +98,13 @@ private function initialize() {
     $this->use_sub_dirs = TRUE;
 
     $customPluginsDir = NULL;
-    if (isset($config->customPHPPathDir)) {
+    if (!empty($config->customPHPPathDir) || $config->customPHPPathDir === '0') {
       $customPluginsDir
         = $config->customPHPPathDir . DIRECTORY_SEPARATOR .
         'CRM' . DIRECTORY_SEPARATOR .
         'Core' . DIRECTORY_SEPARATOR .
         'Smarty' . DIRECTORY_SEPARATOR .
         'plugins' . DIRECTORY_SEPARATOR;
-      if (!file_exists($customPluginsDir)) {
```

Comment:
You still need this line. And with the above change this line only executes if you've specified a custom php dir, and it's something the site owner has control over and wouldn't specify one that was outside open_basedir.

Reply:
There you go.

Wouldn't it be a different world if other people could directly edit a PR. Imagine that!